File: Development/lab6starter/testPokemon.py
from pokemon import Pokemon
import unittest

class testPokemon(unittest.TestCase):
    def __createPokemon(self):
        return Pokemon(1, "Buba", 1, 1, 3)
    
    def test_constructor(self):
        p = self.__createPokemon()
        self.assertEqual(p.name, "Buba")
    
    def test_toJSON(self):
        p = self.__createPokemon()
        strJSON = p.toJSON()
        self.assertEqual(type(strJSON), str)

    def test_toHTMLRow(self):
        p = self.__createPokemon()
        strHTML = p.toHTMLRow()
        self.assertEqual(type(strHTML), str)

if __name__ == '__main__':
    unittest.main()

File: Workspace/Development/lab6starter/testPokemon.js
const Pokemon = require('./pokemon').Pokemon;
const assert = require('assert');

function createPokemon(){
    return new Pokemon(1, "Buba", 1, 1, 3);
}

function test_construtor(){
    let p = createPokemon();
    assert.deepEqual(p.name, "Buba");
}

function test_toJSON(){
    let p = createPokemon();
    let strJSON = p.toJSON();
    assert.equal(typeof(strJSON), 'string');
}

function test_toHTMLRow(){
    let p = createPokemon();
    let strHTML = p.toHTMLRow();
    assert.equal(typeof(strHTML), 'string');
}

const testFunctions = [
    test_construtor,
    test_toJSON,
    test_toHTMLRow
].forEach(func => {
    func.call();
});